regression/cuda/test_float_all_ops.c

```diff
@@ -0,0 +1,437 @@
+
+	rc = hb_mc_device_malloc(&device, N * sizeof(uint32_t), &A_device);
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to allocate memory on device.\n");
+		return rc;
+	}
+
+
+	rc = hb_mc_device_malloc(&device, N * sizeof(uint32_t), &B_device);
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to allocate memory on device.\n");
+		return rc;
+	}
+
+
+	rc = hb_mc_device_malloc(&device, N * sizeof(uint32_t), &res_add_device);
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to allocate memory on device.\n");
+		return rc;
+	}
+
+
+	rc = hb_mc_device_malloc(&device, N * sizeof(uint32_t), &res_sub_device);
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to allocate memory on device.\n");
+		return rc;
+	}
+
+
+	rc = hb_mc_device_malloc(&device, N * sizeof(uint32_t), &res_mul_device);
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to allocate memory on device.\n");
+		return rc;
+	}
+
+
+	rc = hb_mc_device_malloc(&device, N * sizeof(uint32_t), &res_div_device);
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to allocate memory on device.\n");
+		return rc;
+	}
+
+
+	rc = hb_mc_device_malloc(&device, N * sizeof(uint32_t), &res_compare_device);
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to allocate memory on device.\n");
+		return rc;
+	}
+
+
+	rc = hb_mc_device_malloc(&device, N * sizeof(uint32_t), &res_convert_device);
+	if (rc != HB_MC_SUCCESS) { 
+		bsg_pr_err("failed to allocate memory on device.\n");
+		return rc;
+	}
+
+
+
+        /**********************************************************************/
+        /* Allocate memory on the host for A & B                              */
+        /* and initialize with random values and corner cases.                */
+        /**********************************************************************/
+	float A_host[N]; 
+	float B_host[N]; 
+	
+	// Create every possible combination of corner cases for A & B 
+	for (int y = 0; y < NUM_CORNER_CASES; y++) { 
+		for (int x = 0; x < NUM_CORNER_CASES; x++) { 
+			A_host[y * NUM_CORNER_CASES + x] = corner_cases[x]; 
+			B_host[y * NUM_CORNER_CASES + x] = corner_cases[y]; 
+		}
+	}
+
+	// Create every possible combination of A=corner case and B=random float 
+	for (int i = 0; i < NUM_CORNER_CASES; i ++) { 
+		A_host[NUM_CORNER_CASES * NUM_CORNER_CASES + i] = corner_cases[i];
+		B_host[NUM_CORNER_CASES * NUM_CORNER_CASES + i] = hb_mc_generate_float_rand();
+	}
+	
+	// Create every possible combination of A=random float and B=corner case 
+	for (int i = 0; i < NUM_CORNER_CASES; i ++) { 
+		A_host[NUM_CORNER_CASES * NUM_CORNER_CASES + NUM_CORNER_CASES + i] = hb_mc_generate_float_rand();
+		B_host[NUM_CORNER_CASES * NUM_CORNER_CASES + NUM_CORNER_CASES + i] = corner_cases[i];
+	}
+
+	// Fill the rest of A and B with random floats
+	for (int i = NUM_CORNER_CASES * NUM_CORNER_CASES + 2 * NUM_CORNER_CASES; i < N; i++) { 
+		A_host[i] = hb_mc_generate_float_rand();
+		B_host[i] = hb_mc_generate_float_rand();
+	}
```

Comment:
Isn't "all ops" test also testing cases individual op tests consider? Any particular reason we need individual op tests?

Reply:
The "all ops" test also tries a bunch of corner cases like +- inf, +-nan, division by zero, etc. I guess we can reduce the number of test cases in the "all ops" and have heavy testing in the separate operator tests, but in general you are right, the all ops include all other tests (except expf). @drichmond  what do you think are the separate tests redundant?